### PR TITLE
fixes #4424 - overriden attributes get line-through in the UI

### PR DESCRIPTION
**Pull request by unorthodoxgeek**

### CODE REVIEW

File: app/models/lookup_key.rb [hunk 1]
```diff
@@ -158,6 +158,11 @@ def path_elements
     end
   end
 
+  def overriden_on_host?(host)
+    return false if host.is_a?(Hostgroup)
```
Comment (ohadlevy) [inline app/models/lookup_key.rb]: why ignore hostgroups?

File: app/models/lookup_key.rb [hunk 2]
```diff
@@ -158,6 +158,11 @@ def path_elements
...
+    lookup_values.find_by_match("fqdn=#{host.fqdn}").present?
```
Comment (ohadlevy) [inline app/models/lookup_key.rb]: cant you simply do where lookup_values.detect {|lv| lv.lookup_key_id == id} ? (or similar)

Comment (unorthodoxgeek) [inline app/models/lookup_key.rb]: that would be the same as returning true :)

a lookup key's lookup values have lookup_key_id == id by definition.
what we're looking is "is it overridden by the host we're looking at", not "is there something overriding this somewhere"

File: app/models/lookup_key.rb (hunk 1 above)
Comment (unorthodoxgeek) [inline app/models/lookup_key.rb]: why not? how is this method relevant for hosts?

Comment (orrabin) [inline app/models/lookup_key.rb]: hostgroup also uses `puppetclasses/_class_parameters` but you would need to change the match you are looking for.

**Comment by dLobatog** [comment]
@unorthodoxgeek OK with me after fixing @orrabin's remark and making it work for Host groups, thanks!

File: app/models/lookup_key.rb [hunk 3]
```diff
@@ -158,6 +158,11 @@ def path_elements
     end
   end
 
+  def overriden_on_host?(host)
```
Comment (dLobatog) [inline app/models/lookup_key.rb]: Past participle is overridden (double d), I think there is no need to specify 'on_host' as it can be overridden on host groups (perhaps other places in the future).

**Comment by unorthodoxgeek** [comment]
fixed comments. because Gemfile.lock isn't staged, LDAP tests failed on my machine, so I had to change something regarding that.

**Comment by domcleal** [comment]
Please rebase on develop, LDAP tests were fixed in c86528305233a1acf51f4bae8678cd7b182d2a16 (and unrelated changes should be fixed in their own PR).

**Comment by unorthodoxgeek** [comment]
@domcleal  - rebased.

**Comment by unorthodoxgeek** [comment]
[test]

File: app/models/concerns/nested_ancestry_common.rb [hunk 4]
```diff
@@ -30,7 +30,7 @@ def title
 
   def get_title
     return name if ancestry.empty?
-    ancestors.map { |a| a.name + '/' }.join + name
+    ancestors.map { |a| a.name + '/' }.join + name.to_s
```
Comment (ohadlevy) [inline app/models/concerns/nested_ancestry_common.rb:33]: is this required? can name not be string?

File: app/models/lookup_key.rb [hunk 5]
```diff
@@ -162,6 +162,11 @@ def contains_erb?(value)
     value =~ /<%.*%>/
   end
 
+  def overridden?(host)
+    return false unless host.is_a?(Host::Base) || host.is_a?(Hostgroup)
+    lookup_values.find_by_match(host.send(:lookup_value_match)).present?
```
Comment (ohadlevy) [inline app/models/lookup_key.rb:174]: I assume you use send because its private? would it be better to public it?

**Comment by ohadlevy** [comment]
@orrabin can you do another round of tests please? thanks

File: app/models/lookup_key.rb (hunk 5 above)
Comment (unorthodoxgeek) [inline app/models/lookup_key.rb:174]: I think it should be private... this method reaches into host's internals (which I kind of hate, but I see no choice), thus, using send is okay IMHO...

**Comment by orrabin** [comment]
works for me :)

**Comment by dLobatog** [comment]
[test] because of the latest enabled cops

**Comment by unorthodoxgeek** [comment]
@eLobato - I believe this should be ready for merge...

**Comment by dLobatog** [comment]
Merged as 0575b5b0a07c59c443bf2be663cb67d6a5e40f19, thanks @unorthodoxgeek
